File: sciho.py
#!/usr/bin/env python3

min_rating=7.5
min_votes=1000

ratings={
    t: float(r) for (t,r,v)
    in map(lambda x: x.split("\t"), open("title.ratings.tsv").readlines()[1:])
    if float(r) >= min_rating
    if int(v) >= min_votes
}

for t in open("title.basics.tsv").readlines()[1:]:
    (tid,ty,ti,og,ad,y0,y1,rt,gs) = t.split("\t")

    if "Sci-Fi" in gs \
    and ("Horror" in gs or "Thriller" in gs) \
    and ty == "movie" \
    and tid in ratings.keys():
        print(f"{ti} ({y0}), {ratings[tid]}")

File: sciho.exs
#!/usr/bin/env elixir

min_votes = 1000
min_rating = 7.5

ratings = File.stream!("title.ratings.tsv")
|> Stream.drop(1)
|> Enum.reduce(%{}, fn x, acc -> 
  [t, r, v] = x |> String.trim() |> String.split("\t")
  if String.to_float(r) >= min_rating
  and String.to_integer(v) >= min_votes do
    Map.put(acc, t, String.to_float(r))
  else
    acc 
  end end)
    

File.stream!("title.basics.tsv")
|> Stream.drop(1)
|> Stream.map(&String.trim(&1))
|> Stream.map(&String.split(&1, "\t"))
|> Enum.each(fn l ->
  [tid, type, title, _, _, year, _, _, genres] = l
  if type === "movie"
  and genres =~ "Sci-Fi"
  and (genres =~ "Horror" or genres =~ "Thriller")
  and Map.has_key?(ratings, tid) do
    IO.puts("#{title} (#{year}), #{ratings[tid]}")
  end
end)
    
